types/node/index.d.ts

```diff
@@ -2156,37 +2156,29 @@ declare module "child_process" {
 }
 
 declare module "url" {
-    export interface Url {
-        href?: string;
-        protocol?: string;
+    export interface UrlObject {
         auth?: string;
-        hostname?: string;
-        port?: string;
+        hash?: string;
         host?: string;
+        hostname?: string;
+        href?: string;
+        path?: string;
         pathname?: string;
+        port?: string | number;
+        protocol?: string;
+        query?: string | { [key: string]: any; };
         search?: string;
-        query?: string | any;
         slashes?: boolean;
-        hash?: string;
-        path?: string;
     }
 
-    export interface UrlObject {
-        protocol?: string;
-        slashes?: boolean;
-        auth?: string;
-        host?: string;
-        hostname?: string;
-        port?: string | number;
-        pathname?: string;
-        search?: string;
-        query?: { [key: string]: any; };
-        hash?: string;
+    export interface Url extends UrlObject {
+        port?: string;
+        query?: any;
     }
 
     export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
     export function format(URL: URL, options?: URLFormatOptions): string;
-    export function format(urlObject: UrlObject): string;
+    export function format(urlObject: UrlObject | string): string;
```

Comment:
Doesn't `format` use the same type as `parse` returns? If so, how come we need `UrlObject` and `Url`?

Also, I would suggest renaming `Url` to `ParsedURL` to disambiguate it from `URL`

Reply:
From node docu: `The value of urlObject.port is coerced to a string and appended to result`
So in fact everything convertible to string is accepted here (which would be `any`) but you will get a `string | null` from `parse()`.
In real world apps `string | number` is usually what is needed which is most likely the reason why noone complained yet.
There are several such places in node; e.a. you can set everything convertible to `string` as a http header.